### PR TITLE
refactor(x/swaprouter): update swaprouter CLI with osmocli; deprecate comparable gamm CLI

### PR DESCRIPTION
**Pull request by RusAkh**
Closes: #3825

## What is the purpose of the change

Changes cli of `swaprouter` to use osmocli features. Deprecates corresponding cli in `gamm` 

## Documentation and Release Note

not yet

### CODE REVIEW

**Review by p0mvn — APPROVED** [review]
utACK. Thanks. Let's wait for one more before merge

**Comment by p0mvn** [comment]
@RusAkh have you tested each updated CLI command?

It looks like e2e test might be failing

**Review by p0mvn — CHANGES_REQUESTED** [review]
Please look into e2e test and manually test that every CLI command is working

**Comment by RusAkh** [comment]
hello @p0mvn I want to clarify something. Should `create-pool` cmd be in `gamm` or `swaprouter`? It Is currently in both gamm and swaprouter

**Comment by p0mvn** [comment]
> hello @p0mvn I want to clarify something. Should `create-pool` cmd be in `gamm` or `swaprouter`? It Is currently in both gamm and swaprouter

Let's leave it in gamm for now. The reason is that before the e2e upgrade there is no swaprouter module yet. We can change this to swaprouter in the future upgrade.

**Comment by p0mvn** [comment]
Hi @RusAkh . Could you please rebase this PR? Have you figured out the e2e problems?

**Comment by ValarDragon** [comment]
We should not break the gamm cli.

**Comment by RusAkh** [comment]
hello @p0mvn. I did not really find what is wrong with `e2e`. Still looking into that, apologize for the delays

**Comment by RusAkh** [comment]
https://github.com/osmosis-labs/osmosis/actions/runs/3829193643/jobs/6515616760 the error is pretty weird. Another notional team member had the same weird test fail, so I do not think that it is related to changes in this PR

**Comment by p0mvn** [comment]
Can you try merging `main` into your branch please? There was an e2e failure in main which is now resolved

**Comment by p0mvn** [comment]
> We should not break the gamm cli.

@RusAkh based on request, could you please restore the gamm CLI?

**Comment by RusAkh** [comment]
> > We should not break the gamm cli.
> 
> @RusAkh based on request, could you please restore the gamm CLI?

sure

**Comment by RusAkh** [comment]
@p0mvn in this [commit](https://github.com/osmosis-labs/osmosis/pull/3875/commits/40d5a91af0503b7f57f758ad6e1026b6866a3c13) I changed usage of `gamm`'s commands to `swaprouter`'s commands in scripts and other stuff. Shall I bring it back as well or it does not really matter?

File: tests/e2e/configurer/chain/commands.go
Comment (p0mvn) [inline tests/e2e/configurer/chain/commands.go:118]:
```suggestion
// docker container exec <container id> osmosisd tx gamm swap-exact-amount-in <tokeinInCoin> <tokenOutMinAmountInt> --swap-route-pool-ids <swapRoutePoolIds> --swap-route-denoms <swapRouteDenoms> --chain-id=<id>--from=<address> --keyring-backend=test -b=block --yes --log_format=json
```

File: tests/e2e/configurer/chain/commands.go
Comment (p0mvn) [inline tests/e2e/configurer/chain/commands.go:121]:
```suggestion
	cmd := []string{"osmosisd", "tx", "gamm", "swap-exact-amount-in", tokenInCoin, tokenOutMinAmountInt, fmt.Sprintf("--swap-route-pool-ids=%s", swapRoutePoolIds), fmt.Sprintf("--swap-route-denoms=%s", swapRouteDenoms), fmt.Sprintf("--from=%s", from)}
```

**Comment by p0mvn** [comment]
> @p0mvn in this [commit](https://github.com/osmosis-labs/osmosis/pull/3875/commits/40d5a91af0503b7f57f758ad6e1026b6866a3c13) I changed usage of `gamm`'s commands to `swaprouter`'s commands in scripts and other stuff. Shall I bring it back as well or it does not really matter?
> 
> UPD: reverted all related to `x/gamm`, lmk about the question above please

We want gamm to be used in e2e because e2e tests logic prior to upgrade and after upgrade.

Prior to upgrade, there is no swaprouter module yet. As a result, e2e would fail.

**Comment by p0mvn** [comment]
Hi @RusAkh . Any updates here? I'd love to help if any challenges to drive this PR to completion

**Comment by RusAkh** [comment]
sorry for the delay @p0mvn, I brought back gamm, e2e passes!

File: go.work.sum
Comment (p0mvn) [inline go.work.sum:262]: Can you please resolve this conflict?

**Review by p0mvn — APPROVED** [review]
utACK once the conflict is resolved.

@RusAkh can you please confirm that CLI changes have been manually tested? Thank you

File: go.work.sum
Comment (RusAkh) [reply go.work.sum]: resolved

**Comment by RusAkh** [comment]
> utACK once the conflict is resolved.
> 
> @RusAkh can you please confirm that CLI changes have been manually tested? Thank you

yes, manually tested

**Review by p0mvn — APPROVED** [review]

**Pull request merged**
